### ISSUE TITLE
bug: KMS Alias Creation Fails to Return Error

### ISSUE THREAD

**Issue by gsilverstein** [localstack/localstack#7494]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

This looks similar to https://github.com/localstack/localstack/issues/6471

I am trying to sign something using KMS for some tests. It seems like doing so using an alias does not work. For example I create a key and an alias like so:

```
# Add a key used for signing urls
aws-cli --endpoint-url=http://localhost:4566 kms create-key \
  --key-usage SIGN_VERIFY \
  --key-spec RSA_4096 


# Add well known alias for key
aws-cli --endpoint-url=http://localhost:4566 kms create-alias \
  --alias-name "some-nice-alias-name" \
  --target-key-id <key id generated above>
```

I can see that this looks to have worked by verifying the key and alias on the CLI

```
aws-cli --endpoint-url=http://localhost:4566 kms list-keys
{
    "Keys": [
        {
            "KeyId": "f7d2d869-f6b8-4977-96ea-5bd70cb0d5f2",
            "KeyArn": "arn:aws:kms:us-east-1:000000000000:key/<someuuid>"
        }
    ]
}
```

and

```
 aws-cli --endpoint-url=http://localhost:4566 kms list-aliases
{
    "Aliases": [
        {
            "AliasName": "census-webform-url-signing-key",
            "AliasArn": "arn:aws:kms:us-east-1:000000000000:alias/some-nice-alias-name",
            "TargetKeyId": "<sameuuid>",
            "CreationDate": "2023-01-13T16:58:52.279782-05:00"
        }
    ]
}
```

however attempting to sign something does not work

```
# Make sure we can sign
aws-cli --endpoint-url=http://localhost:4566 kms sign \
  --cli-binary-format raw-in-base64-out \
  --key-id "alias/some-nice-alias-name" \
  --message 'wwwtestcom' \
  --message-type RAW \
  --signing-algorithm "RSASSA_PSS_SHA_512"
 
```
  
results in 

An error occurred (NotFoundException) when calling the Sign operation: Unable to find KMS alias with name alias/some-nice-alias-name

### Expected Behavior

Would expect output from the last command not the resulting error.

### How are you starting LocalStack?

With a docker-compose file

#### How are you starting localstack (e.g., `bin/localstack` command, arguments, or `docker-compose.yml`)

    docker run localstack/localstack

#### Client commands (e.g., AWS SDK code snippet, or sequence of "awslocal" commands)

```
aws-cli --endpoint-url=http://localhost:4566 kms create-key \
  --key-usage SIGN_VERIFY \
  --key-spec RSA_4096 


aws-cli --endpoint-url=http://localhost:4566 kms create-alias \
  --alias-name "some-nice-alias-name" \
  --target-key-id <key id generated above>

aws-cli --endpoint-url=http://localhost:4566 kms list-keys
{
    "Keys": [
        {
            "KeyId": "f7d2d869-f6b8-4977-96ea-5bd70cb0d5f2",
            "KeyArn": "arn:aws:kms:us-east-1:000000000000:key/<someuuid>"
        }
    ]
}

 aws-cli --endpoint-url=http://localhost:4566 kms list-aliases
{
    "Aliases": [
        {
            "AliasName": "census-webform-url-signing-key",
            "AliasArn": "arn:aws:kms:us-east-1:000000000000:alias/some-nice-alias-name",
            "TargetKeyId": "<sameuuid>",
            "CreationDate": "2023-01-13T16:58:52.279782-05:00"
        }
    ]
}

aws-cli --endpoint-url=http://localhost:4566 kms sign \
  --cli-binary-format raw-in-base64-out \
  --key-id "alias/some-nice-alias-name" \
  --message 'wwwtestcom' \
  --message-type RAW \
  --signing-algorithm "RSASSA_PSS_SHA_512"
```

### Environment

```markdown
- OS: Macos 12.6.2
- LocalStack: latest docker image
```

### Anything else?

I did test using the same using the actual generated key ID and this works. I also attempted this through a BOTO3 client in python and the same resulted.

**Comment by gsilverstein** [comment]
So this may be able to be closed. I have figured out the root of the issue. It's not that aliases cannot be used but I made an error in my command calls which localstack is not responding to:

```

# Add well known alias for key
aws-cli --endpoint-url=http://localhost:4566 kms create-alias \
  --alias-name "some-nice-alias-name" \
  --target-key-id <key id generated above>
```

Should result in an error. If i run this against real aws I get:

An error occurred (ValidationException) when calling the CreateAlias operation: Alias must start with the prefix "alias/". Please see https://docs.aws.amazon.com/kms/latest/developerguide/kms-alias.html

However against localstack it is acceptable. If I structure my initial command correctly it works, but really localstack should be returning an error when this happens not registering the alias with an invalid name. Knowing the scope is pretty simple maybe I can fix also.

**Comment by bentsku** [comment]
Hello @gsilverstein and thanks for your report!

Indeed, it seems the validation is lacking when creating an Alias. If you'd like to contribute, feel free, I'd be glad to shepherd your PR! You can read more about how to contribute in ours docs: https://docs.localstack.cloud/contributing/

**Comment by gsilverstein** [comment]
Awesome I will try to set aside some time this week to read the docs and begin reviewing the code. I'm sure the change is trivial once I find the right places in your codebase. Once I am in a good place I will drop you a message here @bentsku.

**Comment by deepak2431** [comment]
@bentsku Can I work on this issue?

**Comment by bentsku** [comment]
HI @deepak2431, I think no PR has been opened yet, maybe @gsilverstein can confirm?

**Comment by gsilverstein** [comment]
I have not had an opportunity to spend much time on this or fix it. If someone is willing to look of course I have no objections.

**Comment by deepak2431** [comment]
> I have not had an opportunity to spend much time on this or fix it. If someone is willing to look of course I have no objections.

Sure, @bentsku I will make a PR for this. Will ping you on Slack if I need any help :)

**Comment by bentsku** [comment]
Closed with #7826

Thanks for your report and the investigation! The fix is now merged thanks to @deepak2431, and should be available in the next hours.
